src/sqlfluff/core/linter/discovery.py

```diff
@@ -54,13 +50,72 @@ def _check_ignore_specs(
     return None
 
 
-def _load_ignorefile(filepath: str) -> IgnoreSpecRecord:
+def _load_specs_from_lines(
+    lines: Iterable[str], logging_reference: str
+) -> pathspec.PathSpec:
+    """Load the ignore spec from an iterable of lines.
+
+    Raises SQLFluffUserError if unparsable for any reason.
+    """
+    try:
+        return pathspec.PathSpec.from_lines("gitwildmatch", lines)
+    except Exception:
+        _error_msg = f"Error parsing ignore patterns in {logging_reference}"
+        # If the iterable is a Sequence type, then include the patterns.
+        if isinstance(lines, Sequence):
+            _error_msg += f": {lines}"
+        raise SQLFluffUserError(_error_msg)
+
+
+def _load_ignorefile(dirpath: str, filename: str) -> IgnoreSpecRecord:
     """Load a sqlfluffignore file, returning the parsed spec."""
-    dirname = os.path.dirname(filepath)
-    filename = os.path.basename(filepath)
-    with open(filepath) as f:
-        spec = pathspec.PathSpec.from_lines("gitwildmatch", f)
-    return dirname, filename, spec
+    filepath = os.path.join(dirpath, filename)
+    with open(filepath, mode="r") as f:
+        spec = _load_specs_from_lines(f, filepath)
+    return dirpath, filename, spec
+
+
+def _load_configfile(dirpath: str, filename: str) -> Optional[IgnoreSpecRecord]:
+    """Load ignore specs from a standard config file.
+
+    This function leverages the caching used in the config module
+    to ensure that anything loaded here, can be reused later. Those
+    functions also handle the difference between toml and ini based
+    config files.
+    """
+    filepath = os.path.join(dirpath, filename)
+    # Use normalised path to ensure reliable caching.
+    config_dict = load_config_file_as_dict(Path(filepath).resolve())
+    ignore_section = config_dict.get("core", {})
+    if not isinstance(ignore_section, dict):
+        return None  # pragma: no cover
+    patterns = ignore_section.get("ignore_paths", [])
+    if isinstance(patterns, str):
+        patterns = patterns.split(",")
+    elif not patterns or not isinstance(patterns, list):
+        return None
```

Comment:
this probably wants an `else` to catch edge cases and throw an exception

Reply:
No need in this case, not being caught by either of these clauses means it's already a list and so doesn't need any edits. I'll add a comment to that explain that though 👍